llvm/lib/Target/RISCV/RISCVExpandAtomicPseudoInsts.cpp

```diff
@@ -728,6 +807,72 @@ bool RISCVExpandAtomicPseudo::expandAtomicCmpXchg(
   return true;
 }
 
+static Register getGPRPairEvenReg(Register PairedReg) {
+  switch (PairedReg) {
```

Comment:
Can we do

`(PairedReg - RISCV::X0_PD) * 2 + RISCV::X0` or you can probably ask TargetRegisterInfo for the encoding value of PairedReg and add that to RISCV::X0.

Reply:
Why was this resolved? It doesn't look addressed.

Reply:
I will push my local changes later.